add Job naming convention

julienledem opened the pull request.

mobuchowski commented:
I'd think about one sentence or two to address potential confusion (that I had!) that jobs namespaces and dataset namespaces are somehow connected. While job namespace should be unique to the orchestrator instance/environment, job running in that namespace can potentially operate on datasets in all dataset namespaces.

wslulciuc commented on spec/Naming.md:
```diff
@@ -108,13 +108,52 @@ Identifier :
+https://github.com/OpenLineage/OpenLineage/blob/9445948e41b1880a349cb45ea31f2d6f65225ee0/spec/OpenLineage.json#L268-L317
+
+Example:
+{
```
_Minor_: You'll want to use a code block here to keep the json indentation.

julienledem replied:
makes sense

wslulciuc commented on spec/Naming.md:
```diff
@@ -108,13 +108,52 @@ Identifier :
+
+To capture this, a run event provides a `ParentRun` facet, referring to the parent `Job` and `Run`. This allows tracking a recurring job from the root of the schedule it is running for. 
+
+https://github.com/OpenLineage/OpenLineage/blob/9445948e41b1880a349cb45ea31f2d6f65225ee0/spec/OpenLineage.json#L268-L317
```
_Minor_: This links to the `ParentRunFacet` definition, but you'll also want to use the version that it's `main` (not a commit). I would just add the link in the paragraph above:

> To capture this, a run event provides a [`ParentRun`](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) facet, referring ...

julienledem replied:
It's a tradeoff here. I wanted to make sure the line numbers would stay correct by pointing to a fixed commit. main is probably better though.

wslulciuc replied:
Ahh, ok 👍

wslulciuc reviewed (approved):
Left some minot comments, otherwise 👍

julienledem commented:
Per @mobuchowski 's comment, I have updated the text at the top of `Naming.md` and added links from the main spec int the definition of Jobs and Datasets

julienledem commented:
I have also adressed @wslulciuc 's comments

collado-mike commented on spec/Naming.md:
```diff
@@ -108,13 +108,52 @@ Identifier :
+   - Namespace: the namespace is assigned to the airflow instance. Ex: airflow-staging, airflow-prod
+   - Job: each task in a DAG is a job. name: {dag name}.{task name}
+ - Spark:
+   - Namespace: the namespace is provided as a configuration parameter as in airflow. If there's a parent job, we use the same namespace, otherwise it is provided by configuration.
```
I'd move the line about the parent job namespace to the `Parent job run` section

collado-mike replied:
Also, what happens if a client reports a job with a namespace and a parent run id, but the namespace doesn't match the parent run's namespace?

julienledem replied:
In that case, the consumer of the event takes the namespace into account in the fully qualified path, using the same namespace allows for a shorter (and more readable) one.

julienledem commented:
And addressed @collado-mike 's comments
